stan/math/prim/fun/log2.hpp

```diff
@@ -45,12 +28,13 @@ struct log2_fun {
    */
   template <typename T>
   static inline T fun(const T& x) {
```

Comment:
Unrelated to this PR but why are these defined as static?

Reply:
They're static so that there's no object creation/destruction at run time.   Any time you can move something to a static function it's a win.  You lose generality in that we can't accept an arbitrary functor, but we never need to in this application.

Reply:
Cool cool I don't see any difference between making these static vs. inline

https://godbolt.org/z/fIlp6-